File: shared/integration_service_example.py
import requests
from schemas.bid import BidRequest, BidResponse
from schemas.ltv_prediction import LTVPredictionRequest, LTVPredictionResponse

# Example: Service-to-service call (Python)
def call_syncflow_bid():
    req = BidRequest(user_id="u1", ad_slot="banner_top", bid_amount=1.23)
    resp = requests.post("http://syncflow:8000/execute-bid", json=req.dict())
    if resp.status_code == 200:
        bid_resp = BidResponse(**resp.json())
        print("BidResponse:", bid_resp)
    else:
        print("Bid failed:", resp.text)

def call_syncvalue_ltv():
    req = LTVPredictionRequest(user_id="u1", features={"f1": 1.0})
    resp = requests.post("http://syncvalue:8000/predict-ltv", json=req.dict())
    if resp.status_code == 200:
        ltv_resp = LTVPredictionResponse(**resp.json())
        print("LTVPredictionResponse:", ltv_resp)
    else:
        print("LTV failed:", resp.text)

File: shared/integration_service_example.go
package shared

import (
	"bytes"
	"encoding/json"
	"fmt"
	"kiki_agent/schemas"
	"net/http"
)

func CallSyncflowBid() {
	bidReq := schemas.BidRequest{UserID: "u1", AdSlot: "banner_top", BidAmount: 1.23}
	body, _ := json.Marshal(bidReq)
	resp, err := http.Post("http://syncflow:8000/execute-bid", "application/json", bytes.NewReader(body))
	if err != nil {
		fmt.Println("Bid call failed:", err)
		return
	}
	defer resp.Body.Close()
	var bidResp schemas.BidResponse
	if err := json.NewDecoder(resp.Body).Decode(&bidResp); err == nil {
		fmt.Printf("BidResponse: %+v\n", bidResp)
	} else {
		fmt.Println("Bid decode failed:", err)
	}
}

func CallSyncvalueLTV() {
	ltvReq := schemas.LTVPredictionRequest{UserID: "u1", Features: map[string]interface{}{"f1": 1.0}}
	body, _ := json.Marshal(ltvReq)
	resp, err := http.Post("http://syncvalue:8000/predict-ltv", "application/json", bytes.NewReader(body))
	if err != nil {
		fmt.Println("LTV call failed:", err)
		return
	}
	defer resp.Body.Close()
	var ltvResp schemas.LTVPredictionResponse
	if err := json.NewDecoder(resp.Body).Decode(&ltvResp); err == nil {
		fmt.Printf("LTVPredictionResponse: %+v\n", ltvResp)
	} else {
		fmt.Println("LTV decode failed:", err)
	}
}
